lib/node_modules/@stdlib/blas/base/sspr/test/test.ndarray.native.js

```diff
@@ -0,0 +1,520 @@
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+/* eslint-disable max-len */
+
+'use strict';
+
+// MODULES //
+
+var resolve = require( 'path' ).resolve;
+var tape = require( 'tape' );
+var Float32Array = require( '@stdlib/array/float32' );
+var str2enum = require( '@stdlib/blas/base/matrix-triangle-str2enum' );
+var str2enumLayout = require( '@stdlib/blas/base/layout-str2enum' );
+var tryRequire = require( '@stdlib/utils/try-require' );
+
+
+// FIXTURES //
+
+var rl = require( './fixtures/row_major_l.json' );
+var ru = require( './fixtures/row_major_u.json' );
+var rxp = require( './fixtures/row_major_xp.json' );
+var rxn = require( './fixtures/row_major_xn.json' );
+var rox = require( './fixtures/row_major_ox.json' );
+var rsap = require( './fixtures/row_major_sap.json' );
+var rsapn = require( './fixtures/row_major_sapn.json' );
+var roap = require( './fixtures/row_major_oap.json' );
+var rcap = require( './fixtures/row_major_complex_access_pattern.json' );
+
+var cl = require( './fixtures/column_major_l.json' );
+var cu = require( './fixtures/column_major_u.json' );
+var cxp = require( './fixtures/column_major_xp.json' );
+var cxn = require( './fixtures/column_major_xn.json' );
+var cox = require( './fixtures/column_major_ox.json' );
+var csap = require( './fixtures/column_major_sap.json' );
+var csapn = require( './fixtures/column_major_sapn.json' );
+var coap = require( './fixtures/column_major_oap.json' );
+var ccap = require( './fixtures/column_major_complex_access_pattern.json' );
+
+
+// VARIABLES //
+
+var sspr = tryRequire( resolve( __dirname, './../lib/ndarray.native.js' ) );
+var opts = {
+	'skip': ( sspr instanceof Error )
+};
+
+
+// TESTS //
+
+tape( 'main export is a function', opts, function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof sspr, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function has an arity of 10', opts, function test( t ) {
+	t.strictEqual( sspr.length, 10, 'returns expected value' );
+	t.end();
+});
+
+tape( 'the symmetric rank 1 operation `A = α*A*x*x^T + A` (row-major, lower)', opts, function test( t ) {
+	var expected;
+	var data;
+	var out;
+	var ap;
+	var x;
+
+	data = rl;
+
+	ap = new Float32Array( data.AP );
+	x = new Float32Array( data.x );
+
+	expected = new Float32Array( data.AP_out );
+
+	out = sspr( str2enumLayout( data.order ), str2enum( data.uplo ), data.N, data.alpha, x, data.strideX, data.offsetX, ap, data.strideAP, data.offsetAP );
```

Comment:
Why are you providing enums here and below when the implementation takes care of that?